### PR TITLE
dev/allow_assign_user

### PR DESCRIPTION
**Pull request by abierbaum**

### CODE REVIEW

File: src/app/issue/issue.tpl.html [hunk 1]
```diff
@@ -10,14 +10,9 @@
...
     </span>
-    <img ng-if="issueCtrl.getAssignedUser()"
-         class="avatar"
-         title="Assigned to {{issueCtrl.getAssignedUser().login}}"
-         ng-src="{{issueCtrl.getAssignedUser().avatar_url}}?s=30" />
-    <img ng-if="!issueCtrl.getAssignedUser()"
-         class="avatar empty"
-         title="Assigned to no one."
-         src="http://www.gravatar.com/avatar/00000000000000000000000000000000?d=mm&f=y">
+    <img class="avatar"
```
Comment (abierbaum) [inline src/app/issue/issue.tpl.html:13]: Took this down to just one image to keep things simpler and moved the logic for what to use for the given user into the controller.  (yes, possible to do in the template, but seems cleaner to do it in controller)

Comment (my8bird) [inline src/app/issue/issue.tpl.html:13]: I am not wild about either of the options we have tried so far here.

File: src/app/issue/issue_ctrl.js [hunk 2]
```diff
@@ -24,8 +24,23 @@ mod.controller('IssueCtrl', function($scope, $modal, $rootScope, trRepoModel) {
...
+            return {
+               name       : this.issue.assignee.login,
+               avatar_url : this.issue.assignee.avatar_url + "?s=" + avSize
+            };
+         } else {
+            return {
+               name       : 'no one',
+               avatar_url : 'http://www.gravatar.com/avatar/0?d=mm&f=y&s=' + avSize
+            };
+         }
```
Comment (my8bird) [inline src/app/issue/issue_ctrl.js:43]: What this was a directive would that clean up the template and keep this template logic out of the controller?

``` html
<tr-avatar ng-model="issueCtrl.issue.assignee" />
```

File: src/app/issue/issue_ctrl.js [hunk 3]
```diff
@@ -54,8 +69,10 @@ mod.controller('IssueCtrl', function($scope, $modal, $rootScope, trRepoModel) {
       showIssueDetails: function() {
          // Create a local scope for the template and add the issue into it
          var modal_scope = $rootScope.$new();
+         modal_scope.$id = "modal:issue_details:" + modal_scope.$id;
```
Comment (my8bird) [inline src/app/issue/issue_ctrl.js:72]: _np_ I think you have been doing this $is assignment in the controllers not externally most places

File: src/app/issue/issue_ctrl.js [hunk 4]
```diff
@@ -66,6 +83,22 @@ mod.controller('IssueCtrl', function($scope, $modal, $rootScope, trRepoModel) {
...
          $modal.open(opts);
+      },
+
+      /**
+      * Called to assign the given user to the issue.
+      */
+      assignUser: function(user) {
+         console.log("Assigning: " + user.login);
+         // short-circut locally to get immediate update
+         this.issue.assignee = user;
```
Comment (my8bird) [inline src/app/issue/issue_ctrl.js:94]: Me likey

File: src/common/github_api/github_api.js [hunk 5]
```diff
@@ -132,21 +132,21 @@ angular.module('github.api', ['restangular'])
 
    /**
     @ngdoc    function
-    @name     listRepoUsers
+    @name     listRepoAssignees
     @methodOf github.api.gh
 
     @description
-    Returns the list if users which have access to the supplied repository
+    Returns the list if users which can be assigned to an issue
```
Comment (my8bird) [inline src/common/github_api/github_api.js:139]: `if` -> `of`

**Comment by my8bird** [comment]
+1, but if you have a few seconds looking at an avatar directive would be nice

**Pull request merged**